src/components/FeatureStyle/FeatureStyle.js

```diff
@@ -0,0 +1,709 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { Feature } from 'ol';
+import { Stroke, Style, Icon } from 'ol/style';
+import { asString } from 'ol/color';
+import Select from '../Select';
+import Button from '../Button';
+
+const propTypes = {
+  /**
+   *  The feature to modify.
+   */
+  feature: PropTypes.instanceOf(Feature),
+
+  /**
+   * Index of the style to modify.
+   * Only necessary feature.getStyleFunction() returns an array of styles.
+   */
+  styleIdx: PropTypes.number,
+
+  /**
+   * List of colors available for modifcation.
+   */
+  colors: PropTypes.arrayOf(
+    PropTypes.shape({
+      label: PropTypes.string,
+      value: PropTypes.array,
+      border: PropTypes.string,
+    }),
+  ),
+
+  /**
+   * List of sizes available for text modification.
+   */
+  textSizes: PropTypes.arrayOf(
+    PropTypes.shape({
+      label: PropTypes.string,
+      scale: PropTypes.number,
+    }),
+  ),
+
+  /**
+   * List of sizes available for icon modification.
+   */
+  iconSizes: PropTypes.arrayOf(
+    PropTypes.shape({
+      label: PropTypes.string,
+      value: PropTypes.array,
+      scale: PropTypes.number,
+    }),
+  ),
+
+  /**
+   * List of categories of icons available for modification.
+   */
+  iconCategories: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.string,
+      useColorOption: PropTypes.bool,
+      type: PropTypes.oneOf(['css', 'img']),
+      regex: PropTypes.regex,
+      icons: PropTypes.array,
+    }),
+  ),
+
+  /** Translation function */
+  t: PropTypes.func,
```

Comment:
It will be better but in that case we have title to translate in the render method: 'Modify icon size',....
I don't want to pass all the texts in properties.

Reply:
Excuse accepted :+1: